specification/04-requirements.adoc

```diff
@@ -150,11 +150,31 @@ In order to protect direct access to confidential memory, a CoVE-IO device must
 be attached to a PCIe Root Port that is bound to a platform IOMMU. All inbound
 traffic from a TDI must then be translated by the upstream IOMMU.
 
-As the confidential Domain Security Manager (DoSM), the TSM is responsible for
-setting TDI-specific DMA mapping, MSI page tables, and also for configuring
-platform IOMMUs own MSIs. As a consequence, all IOMMUs on a CoVE-IO platform
-must provide a domain isolated Register Programming Interface (RPI) that is
-exclusively accessible to the TSM.
+As Supervisor Domain Security Managers (SDSM), TSMs are responsible for
+setting TDI-specific DMA mappings, MSI page tables, and also for configuring
+platform IOMMUs own MSIs. As a consequence, IOMMU instances on a CoVE-IO
+platform may be directly and exclusively tied to a TSM controlled supervisor
+domain. allowing the TSM to manage address translation and protection for DMA
+that originates from devices under its control.
+
+Each IOMMU offers a memory-mapped Register Programming Interface (RPI) that the
+associated TSM uses to configure and control the IOMMU. The RDSM, through the
+<<Smmmtt>> extension, enforces that this interface is access-restricted to the
+TSM the IOMMU is assigned to. This guarantees that a TSM currently running on a
+RISC-V hart has exclusive access to the physical address of its IOMMU RPI.
+
+On platforms supporting the MTT and IO-MTT <<Smmtt>> extensions, any PCIe device
+(regardless of its CoVE-IO compliance) can be assigned to any supervisor domain,
+as long as this domain has RDSM-enforced exclusive access to an IOMMU instance
+RPI.
+
+Non CoVE-IO-compliant devices might be assigned to TSM controlled supervisor
```

Comment:
what is definition of "Non CoVE-IO-compliant"? No IDE? No TDISP?
Why we need to mention that?

Reply:
> what is definition of "Non CoVE-IO-compliant"? No IDE? No TDISP? 

Sorry, that does not make a lot of sense. I meant TEE-IO compliant. I fixed the wording now.

> Why we need to mention that?

It's part of Qualcomm's comments. They're asking if assigning non TEE-IO devices to a TVM is possible/supported.

Reply:
It is absolutely possible.
In CoVE spec (not CoVE-IO), any non-TEE-IO device can be assigned to TVM, as long as TVM understands that the communication is not secure, and uses bounce buffer for the communication.

TSM does not need to be involved here at all.
That is why I am confused on why we need to mention in this spec, because I assume this spec is to define architecture for TEE-IO, but not non-TEE-IO.